### ISSUE TITLE
SelectOneMenu: Chrome/Firefox Autofill/Autocomplete

### ISSUE THREAD

**Issue by laurinmueller** [primefaces/primefaces#2424]
## 1) Environment
- PrimeFaces version: 6.1
- Does it work on the newest released PrimeFaces version? No
- Does it work on the newest sources in GitHub? I' don't know
- Application server + version: Websphere 8.5.5
- Affected browsers: Firefox and Chrome (All Browser with Autofill Form functionality)

## 2) Expected behavior
I add a Autofill Profile in Chrome with an adress with a country( for example Switzerland, see printscreen).

![image](https://user-images.githubusercontent.com/1231069/26926465-1df41ae6-4c4e-11e7-86c7-c82e4b3e8c82.png)

After I saved my Autofill Profile, i start fill in a primefaces address form . After I start typing my family-name ("Shaw)  the autofill feature kicks in and fills all other fields of the form(given-name, street, country).
![image](https://user-images.githubusercontent.com/1231069/26927271-78ee48d8-4c51-11e7-8b0b-9371185dec6b.png)

## 3) Actual behavior

I add a Autofill Profile in my Browser with an adress with a country( for example Switzerland, see printscreen).

![image](https://user-images.githubusercontent.com/1231069/26926465-1df41ae6-4c4e-11e7-86c7-c82e4b3e8c82.png)

After I saved my Autofill Profile, i start fill in a primefaces address form . After I start typing my family-name ("Shaw)  the autofill feature kicks in and fills all other fields of the form(given-name, street). BUT the Country selectOneMenu still shows the firstEntry of the country list(germany), but expected would be "Switzerland".... alltough when I submit the form, the correct value is submitted(Switzerland)

The Autofill feature of the browser modifies only the hidden HTML-Select and the SelectOneMenu component does not get informed by that change and doesn't change the label.

![image](https://user-images.githubusercontent.com/1231069/26927255-5da7593e-4c51-11e7-9d00-12f6bd9a19b6.png)

## 4) Steps to reproduce
Create a autofill profile in chrome as shown in the Printscreen:

![image](https://user-images.githubusercontent.com/1231069/26926465-1df41ae6-4c4e-11e7-86c7-c82e4b3e8c82.png)

Then goto bug.xhtml and start typing "Shaw" in the family name field. 

The Autofill fills the other values. The Country dropdown shows Germany. Press Submit, the submitted value altough is correct(switzerland)

## 5) Sample XHTML

bug.xhtml

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:f="http://java.sun.com/jsf/core"
      xmlns:h="http://java.sun.com/jsf/html"
      xmlns:p="http://primefaces.org/ui">

    <h:head>
        <title>Autofill and SelectOneMenu Bug</title>
    </h:head>
    <h:body>
    	<h2>Autofill and SelectOneMenu Bug</h2>
    	<h:form>
    	
    	<p:panelGrid columns="2">
    						<p:column>
    	   					<p:outputLabel for="family-name" value="family-name"/>
								</p:column>
								<p:column>
								<p:inputText id="family-name" 
									value="#{myBean.familyName}" required="true" autocomplete="family-name"/>
							</p:column>

							<p:column>
								<p:outputLabel for="given-name" value="given-name"/>
								</p:column>
								<p:column>
								<p:inputText id="given-name" required="true"
									value="#{myBean.givenName}" autocomplete="given-name"/>
							</p:column>

							<p:column>
								<p:outputLabel for="address-line1" value="street"/>
								</p:column>
								<p:column>
								<p:inputText id="address-line1" required="true"
									value="#{myBean.street}" autocomplete="address-line1"/>
							</p:column>
							
							<p:column>
								<p:outputLabel for="country" value="country"/>
							</p:column>
							<p:column>
								<p:selectOneMenu
									id="country"
									value="#{myBean.country}" required="true">
									<f:selectItems value="#{myBean.countries}"/>
									<!-- <p:ajax update="@this"/>   -->
									<!-- THE LINE ABOVE SOLVES THE PROBLEM -->
								</p:selectOneMenu>
							</p:column>

						</p:panelGrid>
						<p:commandButton value="Submit" update="submittedValues"> </p:commandButton>
						
    	</h:form>
    	<p:panel header="submitted Values" id="submittedValues">
    		<h:outputText value="familyName: #{myBean.familyName}"></h:outputText><br/>
    		<h:outputText value="given-name #{myBean.givenName}"></h:outputText><br/>
    		<h:outputText value="street #{myBean.street}"></h:outputText><br/>
    		<h:outputText value="country #{myBean.country}"></h:outputText><br/>
    	</p:panel>
    </h:body>
</html>
```
## 6) Sample bean

MyBean.java

```
import java.util.Arrays;
import java.util.List;

import javax.faces.bean.ManagedBean;
import javax.faces.bean.ViewScoped;

@ManagedBean
@ViewScoped
public class MyBean {

    private String country;
    private String familyName;
    private String givenName;
    private String street;

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getFamilyName() {
        return familyName;
    }

    public void setFamilyName(String familyName) {
        this.familyName = familyName;
    }

    public String getGivenName() {
        return givenName;
    }

    public void setGivenName(String givenName) {
        this.givenName = givenName;
    }

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public List<String> getCountries(){
        return Arrays.asList("Germany", "Switzerland", "Italy", "France", "USA", "United Kingdom", "India");
    }


}
```

**Comment by tandraschko** [comment]
As i don't think this has an high prio for us, it would be great if you could more analyze the problem, maybe provide a PR or go PrimeFaces PRO to sponsor the fix.

**Comment by melloware** [comment]
I have spent a lot of time investigating this issue.  The problems include:
-  AutoFill is not an HTML5 standard and browsers can choose to respect it how they want to.
-  SelectOneMenu is a custom drawn component with DIV and CSS and a hidden input field where as the browser expects a Text or Select HTML5 element to be able to autofill
- There is no clever trick using a hidden input field to allow it to accept AutoFill etc.
- No browser hooks to the AutoFill feature to even try and capture with JS

So I am afraid the answer here is WontFix as the way the SelectOne was implemented by PrimeTek won't allow it.  You workaround is to use a standard JSF `<h:selectOneMenu>`
